evm/vm/base.py

```diff
@@ -467,12 +476,9 @@ def get_state(cls, chaindb, block, prev_hashes, gas_used):
     def state(self):
         """Return current state property
         """
-        gas_used = 0
-        if self.receipts:
-            gas_used = self.receipts[-1].gas_used
         return self.get_state(
             chaindb=self.chaindb,
             block=self.block,
             prev_hashes=self.previous_hashes,
-            gas_used=gas_used,
+            gas_used=self.block.header.gas_used,
```

Comment:
IIUC, before this change, on every call to `VM.apply_transaction()`, the `VM.state` would include the `gas_used` up to the last applied tx, but now it won't as `self.block.header.gas_used` is only updated at the end of `VM.import_block()`. Is that ok or can it be a problem when applying transactions that would exceed the block's gas limit?

Reply:
I'll add a test to check.

Reply:
This is currently handled.  `State.apply_transaction` calls `State.add_transaction` which updates the block header's `gas_used`.  And `VM.apply_transaction` updates `self.block` on each call.

Reply:
Then in import_block() above we don't need to update `gas_used` when we do [self.block.copy()](https://github.com/ethereum/py-evm/pull/598/files#diff-7a66be18d19df3f35fdebb216454a46fR166)?

Reply:
+1

I'll get this cleaned up in a subsequent PR.  adding a comment.